### ISSUE TITLE
PERMISSION_OS_ERR on Android SOURCE_STORAGE check

### ISSUE THREAD

**Issue by marcanw** [myflashlab/PermissionCheck-ANE#5]
I can't make it work on Android (android 6.0, air 24.0.0.177 and PermissionCheck V2.0.2)
When I try to check SOURCE_STORAGE on Android,  it returns PERMISSION_OS_ERR.

Here is my test code:
trace("showPhotoHelper permissionCheck", gvars.photoPerm, permissionCheck.check(gvars.photoPerm));

Output:
showPhotoHelper permissionCheck onStoragePermissionState 0

Is it normal?

**Comment by marcanw** [comment]
Hello?

**Comment by myflashlab** [comment]
You should pass the needed source to the check parameter:

_ex.check(PermissionCheck.SOURCE_STORAGE)

Try that.

**Comment by marcanw** [comment]
It's already done
if(permissionCheck) gvars.photoPerm = gvars.isIOS ? PermissionCheck.SOURCE_PHOTOS : PermissionCheck.SOURCE_STORAGE;

**Comment by marcanw** [comment]
One week to receive a reply that shows that you don't really read the post is a bit long.

**Comment by myflashlab** [comment]
```PERMISSION_OS_ERR``` happens only if you are trying to access a resource on a wrong platform. maybe your ```gvars.isIOS``` is not detecting the operating system correctly? instead try using the [os](http://myflashlab.github.io/asdoc/com/myflashlab/air/extensions/nativePermissions/PermissionCheck.html#os) property of the ANE.

**Comment by marcanw** [comment]
You can see that the trace shows that the it is detecting the system correctly. It was tested on Android you it shows "onStoragePermissionState"

Here is my test code:
trace("showPhotoHelper permissionCheck", gvars.photoPerm, permissionCheck.check(gvars.photoPerm));

Output:
showPhotoHelper permissionCheck onStoragePermissionState 0

**Comment by myflashlab** [comment]
please send us your manifest + the whole AS function which you are using so we can look into this more. 
while doing that, please trace the [os](http://myflashlab.github.io/asdoc/com/myflashlab/air/extensions/nativePermissions/PermissionCheck.html#os) property and confirm that it outputs ```android```?

**Comment by marcanw** [comment]
Here is a simple code:
permissionCheck = new PermissionCheck();
trace("permissionCheck", permissionCheck.os, permissionCheck.check(PermissionCheck.SOURCE_STORAGE));

Output:
permissionCheck android 0

Manifest attached

[scrollingtile-app.zip](https://github.com/myflashlab/PermissionCheck-ANE/files/631079/scrollingtile-app.zip)

**Comment by marcanw** [comment]
16 days and I still not have a valid answer. This is a really bad support.
I never succeed to use this ane on android and I didn't receive the support to find a solution.
There was several ane that was appealing in your collection but according to my bad experience I don't think I will buy them.

**Comment by myflashlab** [comment]
Unfortunately we are not able to duplicate the problem. and no one else is reporting something similar. unless you are able to send us a demo project to duplicate the problem, we are not sure how we can fix the problem!

**Comment by myflashlab** [comment]
And by the way, if you are using the latest version of the ANE, V2.0.2 you need to add the required dependencies:

```xml
    <!-- The following dependency ANEs are only required when compiling for Android -->
    <extensionID>com.myflashlab.air.extensions.dependency.androidSupport</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.overrideAir</extensionID>
```

You don't seem to have added them in your manifest.

**NOTE:** to help developers with the dependencies, we have developed an [ANE inspector tool](http://www.myflashlabs.com/a-tool-to-make-ane-integration-into-your-air-apps-a-breeze/)

**Comment by marcanw** [comment]
I can't use androidSupport and overrideAir. I receive this message during the compilation.
![image](https://cloud.githubusercontent.com/assets/21979056/21094028/33401376-c055-11e6-97ec-401395a52bf1.png)

I also tried the inspector (without the androidSupport and without the overideAir ane) and I got no error.
Here is the code:
_if (!Inspector.check(PermissionCheck, true, true))
{
    // if the check method returns false, it means that the target ANE can't run
    // on this platform. in this case, you should avoid initializing the target ANE
    trace("Inspector.lastError = " + Inspector.lastError);
}
else trace("Inspector no error " + Inspector.VERSION);_

And the output:
_Inspector no error 1.0.14_

**Comment by myflashlab** [comment]
Ok, this is not making any sense now. The error message says something about com.facebook..... but we are talking about the PermissionCheck ANE.

If you are using a Facebook ANE from other ANE providers, we suggest you to switch to our [Facebook ANE](https://github.com/myflashlab/facebook-ANE/) instead because the error message seems to tell that the compiler is having problem with .class files as they are not using a modern compiler... On the other hand we have kept all our ANEs updated with the latest versions.

Anyway, let's get back to the main issue: We just recompiled the sample codes and can surely confirm that the inspector tool can correctly detect missing dependencies for the PermissionCheck ANE.

Try this: Create a simple demo project with the problem and send us the files here (do NOT include the licensed ANEs of course). Maybe this way we can see why you are receiving such wired problems. And in your demo project, please do not include any ANEs from other ANE providers because we don't have access to those sources and in result we can't provider any support for their possible conflicts with our ANEs.

**Comment by myflashlab** [comment]
plz let us know how's things going on this matter?

**Comment by marcanw** [comment]
Someone made an ane for me few month ago that solved the problem for the Android Permission.
I wanted to rationalise the ane by using only yours (I'm already use it for iOS) but now I don't have the time to look into this, so I'll keep it that way for the moment.

**Comment by myflashlab** [comment]
Ok, feel free to open a new ticket later if required.
